File: sample/src/main/java/ru/terrakok/cicerone/sample/mvp/bottom/BottomNavigationPresenter.java
package ru.terrakok.cicerone.sample.mvp.bottom;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpPresenter;
import com.github.terrakok.cicerone.Router;

/**
 * Created by terrakok 25.11.16
 */
@InjectViewState
public class BottomNavigationPresenter extends MvpPresenter<BottomNavigationView> {
    private Router router;

    public BottomNavigationPresenter(Router router) {
        this.router = router;
    }

    public void onBackPressed() {
        router.exit();
    }
}

File: sample/src/main/java/com/github/terrakok/cicerone/sample/mvp/bottom/BottomNavigationPresenter.kt
package com.github.terrakok.cicerone.sample.mvp.bottom

import com.github.terrakok.cicerone.Router
import moxy.InjectViewState
import moxy.MvpPresenter

/**
 * Created by terrakok 25.11.16
 */
@InjectViewState
class BottomNavigationPresenter(
        private val router: Router
) : MvpPresenter<BottomNavigationView>() {

    fun onBackPressed() {
        router.exit()
    }
}